sources/Algo/src/Algo.cpp

```diff
@@ -169,12 +169,22 @@ ControllerInterfaceDefinitionAlgorithm::ControllerInterfaceDefinitionAlgorithm(s
 void
 ControllerInterfaceDefinitionAlgorithm::operator()(const NodePtr& node) {
   for (const auto& converter : node->converterInterfaces) {
-    auto hvdcLine = converter.hvdcLine;
+    const auto& hvdcLine = converter.hvdcLine;
+    bool inserted;
+    std::tie(std::ignore, inserted) = hvdcLinesIds_.emplace(hvdcLine->id);
     HvdcLineDefinition::Position position;
     if (converter.converterId == hvdcLine->converter1_id) {
-      position = HvdcLineDefinition::Position::FIRST_IN_MAIN_COMPONENT;
+      if (!inserted) {
+        position = HvdcLineDefinition::Position::BOTH_IN_MAIN_COMPONENT;
+      } else {
+        position = HvdcLineDefinition::Position::FIRST_IN_MAIN_COMPONENT;
+      }
     } else if (converter.converterId == hvdcLine->converter2_id) {
-      position = HvdcLineDefinition::Position::SECOND_IN_MAIN_COMPONENT;
+      if (!inserted) {
+        position = HvdcLineDefinition::Position::BOTH_IN_MAIN_COMPONENT;
```

Comment:
With this code, we will add twice the same line with 2 different positions in hvdcLines_, correct ? Is this something we want ?

Reply:
No we don't want that, I will modify it